### DISCUSSION TITLE
Bosch seat motor counter

### DISCUSSION THREAD

**Discussion by IsaacLiimatta** [frc5567/2023-Robot#150] [Team Posts]
Here's an example of code for the Bosch seat motors counter (if its necessary)
```
import edu.wpi.first.wpilibj.DigitalInput;
import edu.wpi.first.wpilibj.Counter;
DigitalInput seatMotor = new DigitalInput (0);
Counter motorCounter = new Counter (seatMotor); `


private int position 0;
    
public void periodic() {
  if (armPosition > 0) {
    position += motorCounter.get();
  } else {
    position -= motorCounter.get();
  }
  motorCounter.reset();
}
```

**Comment by coverbeek** [comment]
The Bosch Seat motor data sheet is here: 
https://andymark-weblinc.netdna-ssl.com/media/W1siZiIsIjIwMTkvMDEvMjIvMTUvNDAvMDgvNDE1ZWE5ZmUtZjA4Ni00M2MxLWE1YjMtYThhNzJhZWE0NmMxLzIwMTYtMTItMjEgc3BlYyBzaGVldCAtIEJvc2NoIEZSQyBtb3RvciA2IDAwNCBSQTMgMTk0LTA2LnBkZiJdXQ/2016-12-21%20spec%20sheet%20-%20Bosch%20FRC%20motor%206%20004%20RA3%20194-06.pdf?sha=42a181c2a13b270e

According to it, the Hall encoder activates on every rotation of the armature, which results in 174.9 ticks per rev.
Also, according to it, our rotation rate on the output shaft is ~24 RPM.
So in order to achieve a 1/4 rotation we need to move 43.725 ticks. We probably won't have that precision, so 44 ticks is probably good. 
Note that 24 RPM is still probably a little fast than we want (full power for 1 second would get us to 144 degrees or about 40% of a rotation), so we will need to use something less than full power or timeslice very precisely.
